src/m1_facilitator/Manifest/Manifest.ts

```diff
@@ -0,0 +1,259 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import fs from 'fs-extra';
+import yaml from 'js-yaml';
+import { Validator as JsonSchemaVerifier } from 'jsonschema';
+import schema from './manifest.schema.json';
+
+interface ManifestInputChain {
+  avatar_account: string;
+  node_endpoint: string;
+  graph_ws_endpoint: string;
+  graph_rpc_endpoint: string;
+}
+
+interface ManifestInputAccount {
+  keystore_path: string;
+  keystore_password_path: string;
+}
+
+interface ManifestInputType {
+  version: string;
+  architecture_layout: string;
+  personas: string[];
+  metachain: {
+    origin: ManifestInputChain;
+    auxiliary: ManifestInputChain;
+  };
+  accounts: Record<string, ManifestInputAccount>;
+  origin_contract_addresses: Record<string, string>;
+  facilitate_tokens: string[];
+}
+
+/** Enum of architecture layouts which facilitator supports. */
+enum ArchitectureLayout {
+  MOSAIC1 = 'M1',
+}
+
+/** Enum of different personas which facilitator supports. */
+enum Personas {
+  FACILITATOR = 'facilitator',
+  VALIDATOR = 'validator'
+}
+
+/**
+ * The class holds database configurations.
+ */
+export class DBConfig {
+  /** Database path */
+  public path: string;
+
+  public constructor() {
+    this.path = '';
+  }
+}
+
+/**
+ * It holds avatar information.
+ */
+export class Avatar {
+  public readonly keystorePath: string;
+
+  public readonly keystorePasswordPath: string;
```

Comment:
should we read password from file instead of path.

Reply:
Done!